frontend/server/src/Controllers/Contest.php

```diff
@@ -107,12 +108,28 @@ public static function apiList(\OmegaUp\Request $r): array {
             $r['participating'],
             'participating'
         );
-
+        $tabName = $r->ensureOptionalEnum(
+            'tab_name',
+            [
+                'current',
+                'future',
+                'past',
+            ]
+        );
         $page = (isset($r['page']) ? intval($r['page']) : 1);
         $pageSize = (isset($r['page_size']) ? intval($r['page_size']) : 20);
         $activeContests = isset($r['active'])
             ? \OmegaUp\DAO\Enum\ActiveStatus::getIntValue(intval($r['active']))
             : \OmegaUp\DAO\Enum\ActiveStatus::ALL;
+        if (!is_null($tabName)) {
+            if (strcmp($tabName, 'current')  == 0) {
+                $activeContests = \OmegaUp\DAO\Enum\ActiveStatus::ACTIVE;
+            } elseif (strcmp($tabName, 'future') == 0) {
+                $activeContests = \OmegaUp\DAO\Enum\ActiveStatus::FUTURE;
+            } elseif (strcmp($tabName, 'past') == 0) {
```

Comment:
```suggestion
            if ($tabName == 'current') {
                $activeContests = \OmegaUp\DAO\Enum\ActiveStatus::ACTIVE;
            } elseif ($tabName == 'future') {
                $activeContests = \OmegaUp\DAO\Enum\ActiveStatus::FUTURE;
            } else {
```

la última condición era de lo que se quejaba psalm. antes ya habíamos aseverado que `$tabName` sólo podía ser uno de `['current', 'future', 'past']`. como en L129 el único valor que puede tener es `'past'`, la prueba es innecesaria!